### DISCUSSION TITLE
PDF to PCL Conversions

### DISCUSSION THREAD

**Discussion by Bharad123** [ImageMagick/ImageMagick#2001] [General]
I was trying to convert pdf to pcl with ImageMagick, But PCL it is not scaling exactly to the page and quality of the file going away. Can someone suggest me with appropriate script?

**Comment by dlemstra** [comment]
Please explain in more detail what you are doing now and include some images that show what you mean. I suspect that specifying the -density should help you a quality.

**Reply by Bharad123** [reply]
[Original-pdf-file.pdf](https://github.com/ImageMagick/ImageMagick/files/4602997/Original-pdf-file.pdf)
Attached was my PDF file and i am trying to convert it to PCL.

These are my recent scripts i used:
1) /convert   sourceFolder/file.pdf    destinationFolder/file.pcl
2) /convert   sourceFolder/p.pdf -resize 800x800  destinationFolder/p.pcl
3) /convert sourceFolder/p.pdf -extent 1240x1750  -gravity center -density 200x200  destinationFolder/p.pcl
4)  /convert sourceFolder/p.pdf -page a4x5x5  destinationFolder/p.pcl
5)  /convert sourceFolder/p.pdf  -resize 3640x4200 -extent 3640x4200 -gravity center -units PixelsPerInch -density 300x300  destinationFolder/p.pdf
6)   /convert sourceFolder/p.pdf -resize 550x800^\>  -gravity North  destinationFolder/p.pcl

None of them is not generating a correct PCL file.

**Reply by Bharad123** [reply]
[pdf-2.pdf](https://github.com/ImageMagick/ImageMagick/files/4603030/pdf-2.pdf)
This is another example pdf.

**Comment by urban-warrior** [comment]
Add -density before the source image, e.g. `convert -density 300 p.pdf p.pcl`

**Reply by Bharad123** [reply]
Yes. i have added that and that is making a difference in quality it looks better now. BUT, i see the size of generated PCL has been increased abnormally. My Source PDF size was 500KB and resulted PCL size is nearly 4-5 MB. So, is this an expected behavior? or is there any way to reduce the size as well without loosing quality.?

**Comment by fmw42** [comment]
try the following:

`convert -density 288 p.pdf -resize 25% p.pcl`

Nominal density is 72. So 72*4 = 288 and 1/4 = 25%

**Reply by Bharad123** [reply]
I tried that as well. PCL looks too small.
![Result-Pcl](https://user-images.githubusercontent.com/49654386/81494285-4ed69e00-926d-11ea-92f3-bac201f5fd8f.PNG)

**Reply by fmw42** [reply]
Then either increase the density or resize to a larger percent.

**Reply by Bharad123** [reply]
Yes. I did that..
/usr/bin/convert -density 300  /sourcefolder/p.pdf -resize 75% /destinatonfolder/p.pcl.
But It is not at all scaling correctly as you can see it in attached image. It has lot of white space on the right border. I even tried with -gravity option.
I = tried with -border option as well but no luck..
![Updated-Pcl-result](https://user-images.githubusercontent.com/49654386/81495471-16878d80-9276-11ea-8e31-a56df956d928.PNG)

**Comment by dlemstra** [comment]
There are some "fit page" options available for both PDF and PCL. What happens with the following command:

`magick -density 300 -define pdf:fit-page=a4  /sourcefolder/p.pdf -define pcl:fit-to-page=true /destinationfolder/p.pcl`

**Reply by Bharad123** [reply]
Nope. Still not working..
I tried with both of these commands.
1) /usr/bin/convert -density 300 -define pdf:fit-page=a4  /sourcefolder/p.pdf -define pcl:fit-to-page=true  /destinationfolder/p.pcl
2) /usr/bin/convert -density 300 -define pdf:fit-to-page=true
![PCL](https://user-images.githubusercontent.com/49654386/81496066-754f0600-927a-11ea-8fb0-c8b067374e43.PNG)
  /sourcefolder/p.pdf -define pcl:fit-to-page=true  /destinationfolder/p.pcl

**Reply by fmw42** [reply]
Which is the PDF that has problems? If not provided already, please post it.

**Reply by Bharad123** [reply]
Attched PDF.
[pdf.pdf](https://github.com/ImageMagick/ImageMagick/files/4606524/pdf.pdf)

**Reply by fmw42** [reply]
I do not see any issue using IM 6.9.10.10 Q16 Mac OSX Sierra 

` convert -density 300 pdf.pdf x.png`

![x](https://user-images.githubusercontent.com/13458218/81507660-bd3e4f00-92b3-11ea-9a87-0227ccbacc56.png)

What is your version of Ghostscript?  I was using GS 9.50

**Reply by Bharad123** [reply]
I am trying to convert that pdf to PCL not a PNG.

**Reply by fmw42** [reply]
Sorry, I have no tools that will display the resulting PCL file to check it or I do not have the requisite delegate to convert to it.
